pkg/preference/preference.go

```diff
@@ -0,0 +1,436 @@
+	if c.ActiveApplications == nil {
+		c.ActiveApplications = []ApplicationInfo{}
+	}
+
+	for i, app := range c.ActiveApplications {
+		if app.Project == project && c.ActiveApplications[i].ActiveComponent != "" {
+			c.ActiveApplications[i].ActiveComponent = ""
+		}
+	}
+
+	// Write the configuration to file
+	err := util.WriteToFile(c.GlobalConfig, c.Filename)
+	if err != nil {
+		return errors.Wrapf(err, "unable to write configuration file")
+	}
+	return nil
+
+}
+
+// UnsetActiveApplication sets the active application as blank of the given project in the configuration file
+func (c *GlobalConfigInfo) UnsetActiveApplication(project string) error {
+	if c.ActiveApplications == nil {
+		c.ActiveApplications = []ApplicationInfo{}
+	}
+
+	for i, cfgApp := range c.ActiveApplications {
+		if cfgApp.Project == project && c.ActiveApplications[i].Active {
+			c.ActiveApplications[i].Active = false
+		}
+	}
+
+	err := util.WriteToFile(c.GlobalConfig, c.Filename)
+	if err != nil {
+		return errors.Wrap(err, "unable to write configuration file")
+	}
+	return nil
+}
+
+// GetActiveComponent if no component is set as current returns empty string
+func (c *GlobalConfigInfo) GetActiveComponent(application string, project string) string {
+	if c.ActiveApplications != nil {
+		for _, app := range c.ActiveApplications {
+			if app.Project == project && app.Name == application && app.Active {
+				return app.ActiveComponent
+			}
+		}
+	}
+	return ""
+}
+
+// GetActiveApplication get currently active application for given project
+// if no application is active return empty string
+func (c *GlobalConfigInfo) GetActiveApplication(project string) string {
+	if c.ActiveApplications != nil {
+		for _, app := range c.ActiveApplications {
+			if app.Project == project && app.Active {
+				return app.Name
+			}
+		}
+	}
+	return ""
+}
+
+// SetActiveApplication set application as active for given project
+func (c *GlobalConfigInfo) SetActiveApplication(application string, project string) error {
+	if c.ActiveApplications == nil {
+		c.ActiveApplications = []ApplicationInfo{}
+	}
+
+	found := false
+	for i, app := range c.ActiveApplications {
+		// if application exists set is as Active
+		if app.Name == application && app.Project == project {
+			c.ActiveApplications[i].Active = true
+			found = true
+			break
+		}
+	}
+
+	// if application doesn't exists, add it as Active
+	if !found {
+		return fmt.Errorf("unable set application %s as active in config, it doesn't exist", application)
+	}
+	// make sure that no other application is active
+	for i, app := range c.ActiveApplications {
+		if !(app.Name == application && app.Project == project) {
+			c.ActiveApplications[i].Active = false
+		}
+	}
+
+	err := util.WriteToFile(c.GlobalConfig, c.Filename)
+	if err != nil {
+		return errors.Wrap(err, "unable to set current application")
+	}
+	return nil
+}
+
+// AddApplication add  new application to the config file
```

Comment:
add -> adds (plus extra space to remove)

Reply:
resolved